lib/smart_proxy_dynflow_core/callback.rb

```diff
@@ -33,13 +33,13 @@ def callback_resource
       end
 
       def ssl_options
-        return {} unless Settings.instance.use_https
-        client_key = File.read  Settings.instance.ssl_private_key
-        client_cert = File.read Settings.instance.ssl_certificate
+        return {} unless URI.parse(Settings.instance.callback_url).scheme == 'https'
```

Comment:
If we're going to decide whether to use http or https based on parsing the uri, we could (and should) drop the `use_https` from config and settings